pkg/registry/common/seturl/server.go

```diff
@@ -19,44 +19,72 @@ package seturl
 
 import (
 	"context"
-
-	"github.com/networkservicemesh/sdk/pkg/registry/core/next"
+	"net/url"
 
 	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/pkg/errors"
+
 	"github.com/networkservicemesh/api/pkg/api/registry"
+
+	"github.com/networkservicemesh/sdk/pkg/registry/core/next"
+	"github.com/networkservicemesh/sdk/pkg/tools/stringurl"
 )
 
-type setMgrServer struct {
-	url string
+type setURLNSEServer struct {
```

Comment:
Do not think that we should use chain elements that have more than one responsibility.  
I would be suggesting to keep seturl as simple as it was before and move the other logic into localbypass registry serever.

Reply:
I suppose that it is one responsibility to swap between NSMgr URL for Registry and NSE URL for NSMgr.